pkg/controller/component/synthesize_component.go

```diff
@@ -247,13 +254,39 @@ func buildAffinitiesAndTolerations(comp *appsv1alpha1.Component, synthesizeComp
 	return nil
 }
 
-// buildVolumeClaimTemplates builds volumeClaimTemplates for component.
 func buildVolumeClaimTemplates(synthesizeComp *SynthesizedComponent, comp *appsv1alpha1.Component) {
 	if comp.Spec.VolumeClaimTemplates != nil {
 		synthesizeComp.VolumeClaimTemplates = toVolumeClaimTemplates(&comp.Spec)
 	}
 }
 
+// limitSharedMemoryVolumeSize limits the shared memory volume size to memory requests/limits.
+func limitSharedMemoryVolumeSize(synthesizeComp *SynthesizedComponent, comp *appsv1alpha1.Component) {
+	shm := defaultShmQuantity
+	if comp.Spec.Resources.Limits != nil {
+		if comp.Spec.Resources.Limits.Memory().Cmp(shm) > 0 {
+			shm = *comp.Spec.Resources.Limits.Memory()
+		}
+	}
+	if comp.Spec.Resources.Requests != nil {
+		if comp.Spec.Resources.Requests.Memory().Cmp(shm) > 0 {
+			shm = *comp.Spec.Resources.Requests.Memory()
+		}
+	}
+	for i, vol := range synthesizeComp.PodSpec.Volumes {
+		if vol.EmptyDir == nil {
+			continue
+		}
+		if vol.EmptyDir.Medium != corev1.StorageMediumMemory {
+			continue
+		}
+		if vol.EmptyDir.SizeLimit != nil && !vol.EmptyDir.SizeLimit.IsZero() {
+			continue
+		}
+		synthesizeComp.PodSpec.Volumes[i].EmptyDir.SizeLimit = &shm
```

Comment:
Here, it should be checked whether the memory empty volume mounted by volumeMount is present before performing the setting. This can avoid unnecessary Pod restarts？